src/me/remie/vulcan/leaguetasks/task/tasks/crafting/CraftLeatherBody.java

```diff
@@ -0,0 +1,173 @@
+        this.shopHelper = new ShopHelper(this);
+        setSkillRequirement(new SkillRequirement(script.ctx, Map.of(SimpleSkills.Skills.CRAFTING, 14)));
+    }
+
+    @Override
+    public void run() {
+        if (hasItemInInventory(NEEDLE_ID) && hasItemInInventory(THREAD_ID) && hasItemInInventory(LEATHER)) {
+            if (!ctx.dialogue.dialogueOpen()) {
+                SimpleItem leather = ctx.inventory.populate().filter(LEATHER).next();
+                SimpleItem needle = ctx.inventory.populate().filter(NEEDLE_ID).next();
+                SimpleItem thread = ctx.inventory.populate().filter(THREAD_ID).next();
+                needle.menuAction("Use");
+                ctx.sleep(600);
+                leather.click(0);
+                ctx.sleep(600);
+            } else {
+                ctx.dialogue.clickDialogueOption(1);
+                ctx.sleep(600);
+            }
+        }
+        if (!hasItemInInventory(NEEDLE_ID)) {
+            shopHelper.buyItem(NEEDLE, ShopType.GENERAL_STORE);
+        }
+
+        if (!hasItemInInventory(THREAD_ID)) {
+            shopHelper.buyItem(THREAD, ShopType.GENERAL_STORE);
+        }
+
+        if (!hasItemInInventory(LEATHER)) {
+            obtainLeather();
+        }
+
+        // Proceed with crafting the leather body if all items are now obtained...
+    }
+
+    private boolean hasItemInInventory(int itemId) {
+        return !ctx.inventory.populate().filter(itemId).isEmpty();
+    }
+
+    private void obtainLeather() {
+        if (hasItemInInventory(THREAD_ID) && hasItemInInventory(NEEDLE_ID)) {
+            // Check inventory for cowhides
+            final int COWHIDE = 1739;
+            if (ctx.inventory.populate().filter(COWHIDE).population() < 1) {
+                // Teleport to cows, kill them, and loot cowhides
+                killCowsAndLootHides();
+            }
+
+            // Check if we have cowhides to tan
+            if (ctx.inventory.populate().filter(COWHIDE).population() >= 1) {
+                // Tan cowhides into leather
+                tanCowhidesIntoLeather();
+            }
+        }
+    }
+
+    private void killCowsAndLootHides() {
+            if (!ctx.pathing.regionLoaded(LeagueScriptConstants.LUMBRIDGE_COWS_REGION_ID)) {
+                if (!teleportHome()) {
+                    return;
+                }
+                if (ctx.teleporter.open()) {
+                    script.setScriptStatus("Teleporting to Cows");
+                    ctx.teleporter.teleportStringPath("Training", "Cows & Goblins");
+                    ctx.onCondition(() -> ctx.pathing.regionLoaded(LeagueScriptConstants.LUMBRIDGE_COWS_REGION_ID), 250, 10);
+                    return;
+                }
+            } else {
+                SimpleGroundItem drop = ctx.groundItems.nearest().peekNext();
+                if (!ctx.groundItems.populate().filterContains("Cowhide").isEmpty() && drop != null && ctx.inventory.canPickupItem(drop)) {
```

Comment:
This feels a bit weird, we cant assume our users won't use ironman. This can easy become awkward when the bot is trying to pickup someone else loot?

Reply:
This is true, i've tried doing this on other servers when i was using IM, I couldn't find a way to prevent this.

Reply:
Never tested it before, but maybe kill npc, grab the position of it and search in that area, around the npc?

Reply:
I think this is fine as we use the shop's for a lot of things and I'm not sure if ironman have access to the items we purchase.

Reply:
👍 good to know. We won't support Ironmans.